### PR TITLE
Fix assmly GUI not working in TGUI TTV

### LINKED ISSUE #2700: Timers don't work on ttvs anymore.
## Round ID:

22531 ever since tgui 3 was merged timers on ttvs dont work anymore

### PR DESCRIPTION
**Pull request by ItsTisme**
## About The Pull Request
closes #2700 which prevented you from opening the assembly GUI when its inside the TTV.

This happened because `ui_status` only checked if the object itself can be interacted with and not the object containing itself. It should work with signaler, proximity and timers aswell.

## Why It's Good For The Game
fix bug good yes

## Changelog
:cl:
fix: assembly's now can be interacted with while inside other objects like TTV
/:cl:

### CODE REVIEW

**Comment by TotallyNotChief** [comment]
does using a ttv in hand open the new tgui3 interface now? because its implemented and it works if you click on a ttv as a ghost

**Comment by ItsTisme** [comment]
> does using a ttv in hand open the new tgui3 interface now? because its implemented and it works if you click on a ttv as a ghost

When you open the TTV, you have a option that says the assmbly name and "remove", this makes it so when you actuely press the assmebly name item you open its GUI and are free to edit is aslong as you have acces to the TTV itself

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Change the actual ui_interacts that are called to have a different default_state, so instead of 
/obj/item/assembly/timer/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, \
									datum/tgui/master_ui = null, datum/ui_state/state = GLOB.hands_state)
you would have GLOB.hands_state as whatever state is required for it

**Comment by ItsTisme** [comment]
> Change the actual ui_interacts that are called to have a different default_state, so instead of
> /obj/item/assembly/timer/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE,
> datum/tgui/master_ui = null, datum/ui_state/state = GLOB.hands_state)
> you would have GLOB.hands_state as whatever state is required for it

I may miss understand what your trying to say here. But assuming you say I make my own `GLOB.hands_state`..

`hands_state` is logic that checks once a UI is opend wether it should remain open. I could make my own version of this but it seems redudent since the `hands_state` checks for alot of logic wether carbons are holding something and making another global value to snowflake assembly hand state seems unneeded

Or iam getting wrong what yout trying to say

**Review by PowerfulBacon — DISMISSED** [review]
This if fine actually

File: code/modules/assembly/assembly.dm [hunk 1]
```diff
@@ -125,3 +125,9 @@
 
 /obj/item/assembly/interact(mob/user)
 	return ui_interact(user)
+
+/obj/item/assembly/ui_status(mob/user)
+	.= ..()
+	if(src.can_interact(user) || holder.can_interact(user))
+		. = max(., UI_INTERACTIVE)
```
Comment (PowerfulBacon) [inline code/modules/assembly/assembly.dm]: Might as well just return UI_INTERACTIVE

Comment (ike709) [inline code/modules/assembly/assembly.dm]:
```suggestion
		return UI_INTERACTIVE
```

File: code/modules/assembly/assembly.dm [hunk 2]
```diff
@@ -125,3 +125,8 @@
 
 /obj/item/assembly/interact(mob/user)
 	return ui_interact(user)
+
+/obj/item/assembly/ui_status(mob/user)
+	.= ..()
```
Comment (ike709) [inline code/modules/assembly/assembly.dm]:
```suggestion
	. = ..()
```

**Pull request merged**